app/controllers/dashboard_controller.rb

```diff
@@ -2,7 +2,7 @@ class DashboardController < ApplicationController
   def show
     @user_tracks = current_user.user_tracks.order(last_touched_at: :desc).limit(3)
     @num_user_tracks = current_user.user_tracks.count
-    @featured_badges = Badge.where(id: current_user.acquired_badges.revealed.order('id desc').limit(4).pluck(:badge_id))
+    @featured_badges = Badge.where(id: current_user.acquired_badges.revealed.pluck(:badge_id)).order(Arel.sql("FIND_IN_SET(rarity, 'legendary,ultimate,rare,common')")).limit(4) # rubocop:disable Layout/LineLength
```

Comment:
```suggestion
    @featured_badges = Badge.where(id: current_user.acquired_badges.revealed.select(:badge_id)).order(Arel.sql("FIND_IN_SET(rarity, 'legendary,ultimate,rare,common')")).limit(4) # rubocop:disable Layout/LineLength
```

I think you want select on the first to use an inner query. I'd break it over 2 lines too.